node/full_node_integration_test.go

```diff
@@ -169,68 +169,110 @@ func TestLazyAggregator(t *testing.T) {
 	}, key, signingKey, proxy.NewLocalClientCreator(app), &cmtypes.GenesisDoc{ChainID: "test", Validators: genesisValidators}, log.TestingLogger())
 	assert.False(node.IsRunning())
 	assert.NoError(err)
-	err = node.Start()
-	assert.NoError(err)
+
+	assert.NoError(node.Start())
 	defer func() {
 		require.NoError(node.Stop())
 	}()
 	assert.True(node.IsRunning())
 
-	require.NoError(err)
-
-	require.NoError(waitForFirstBlock(node.(*FullNode), false))
+	require.NoError(waitForFirstBlock(node.(*FullNode), Header))
 
 	client := node.GetClient()
 
 	_, err = client.BroadcastTxCommit(context.Background(), []byte{0, 0, 0, 1})
 	assert.NoError(err)
-	require.NoError(waitForAtLeastNBlocks(node, 2, false))
+	require.NoError(waitForAtLeastNBlocks(node, 2, Header))
 
 	_, err = client.BroadcastTxCommit(context.Background(), []byte{0, 0, 0, 2})
 	assert.NoError(err)
-	require.NoError(waitForAtLeastNBlocks(node, 3, false))
+	require.NoError(waitForAtLeastNBlocks(node, 3, Header))
 
 	_, err = client.BroadcastTxCommit(context.Background(), []byte{0, 0, 0, 3})
 	assert.NoError(err)
 
-	require.NoError(waitForAtLeastNBlocks(node, 4, false))
+	require.NoError(waitForAtLeastNBlocks(node, 4, Header))
+}
+
+// TestSingleAggregatorTwoFullNodesBlockSyncSpeed tests the scenario where the chain's block time is much faster than the DA's block time. In this case, the full nodes should be able to use block sync to sync blocks much faster than syncing from the DA layer, and the test should conclude within block time
```

Comment:
to make this verify it concludes within the block time, you should use a timer in a go routine. 

```go

// test set up

ch := make(chan struct{})
timer := time.NewTimer(bmConfig.DABlockTime)

go func() {
  	select {
	case <-ch:
		// Channel closed before timer expired.
		return
	case <-timer.C:
		// Timer expired before channel closed.
		t.Fatal("nodes did not sync before DA Block time")
	}
}()


// test

close(ch)
```

Reply:
Thanks for the suggestion. Added this. Although, the linter complains if a goroutine tries to call `t.Fatal` so had to relay the error back through the channel.

Reply:
oh yea, just use `t.Error` with a return instead. Forgot about the linter not liking fatals in a go routine.

Reply:
Modified. Also verified that the test fails successfully if the timer fires too soon.